qlib/kernel/Kernel.rs

```diff
@@ -822,10 +822,14 @@ impl HostSpace {
         return ret;
     }
 
-    pub fn VcpuWait() -> i64 {
-        let mut ret: i64 = 0;
-        HyperCall64(HYPERCALL_VCPU_WAIT, 0, 0, &mut ret as *mut _ as u64, 0);
-        return ret as i64;
+    pub fn VcpuWait() -> TaskId {
+        let mut next: TaskId = TaskId::New(0, 0);
+        HyperCall64(HYPERCALL_VCPU_WAIT, 0, 0, &mut next as *mut _ as u64, 0);
+        
+        assert!(next.PrivateTaskAddr() != 0);
```

Comment:
Also in a general sense, doesn't this make the system panic, when quark is not configured to run with cc?

Perhaps it's better to wrap this with a cfg item?

Reply:
Switches will be added in a future patch